### DISCUSSION TITLE
Inconsistent Data between API and Observational History webpage

### DISCUSSION THREAD

**Discussion by joshwalker9115** [weather-gov/api#374] [Q&A]
**Describe the bug**
I am receiving inconsistent data between what is returned from API calls and what is shown on the NWS observational history page. I am particularly looking at precipitation data however temperature and other values are inconsistent as well.

I have tried double checking metric vs imperial measurement pages and receive "null" where the direct link shows precipitation values. I have found this issue with stations, ID: "KOKV" & "KW99" 

**To Reproduce**
API request:
`https://api.weather.gov/stations/KOKV/observations?limit=300`

returns JSON with several values inconsistent with what is shown on the observational history page:
[https://forecast.weather.gov/data/obhistory/metric/KOKV.html](https://forecast.weather.gov/data/obhistory/metric/KOKV.html)

A particular example as this may be harder to reproduce after several days of no precipitation:
`https://api.weather.gov/stations/KOKV/observations/2020-01-18T15:55:00+00:00`:

```{
  "@context": [
    "https://raw.githubusercontent.com/geojson/geojson-ld/master/contexts/geojson-base.jsonld",
    {
      "wx": "https://api.weather.gov/ontology#",
      "s": "https://schema.org/",
      "geo": "http://www.opengis.net/ont/geosparql#",
      "unit": "http://codes.wmo.int/common/unit/",
      "@vocab": "https://api.weather.gov/ontology#",
      "geometry": {
        "@id": "s:GeoCoordinates",
        "@type": "geo:wktLiteral"
      },
      "city": "s:addressLocality",
      "state": "s:addressRegion",
      "distance": {
        "@id": "s:Distance",
        "@type": "s:QuantitativeValue"
      },
      "bearing": {
        "@type": "s:QuantitativeValue"
      },
      "value": {
        "@id": "s:value"
      },
      "unitCode": {
        "@id": "s:unitCode",
        "@type": "@id"
      },
      "forecastOffice": {
        "@type": "@id"
      },
      "forecastGridData": {
        "@type": "@id"
      },
      "publicZone": {
        "@type": "@id"
      },
      "county": {
        "@type": "@id"
      }
    }
  ],
  "id": "https://api.weather.gov/stations/KOKV/observations/2020-01-18T15:55:00+00:00",
  "type": "Feature",
  "geometry": {
    "type": "Point",
    "coordinates": [
      -78.15,
      39.15
    ]
  },
  "properties": {
    "@id": "https://api.weather.gov/stations/KOKV/observations/2020-01-18T15:55:00+00:00",
    "@type": "wx:ObservationStation",
    "elevation": {
      "value": 222,
      "unitCode": "unit:m"
    },
    "station": "https://api.weather.gov/stations/KOKV",
    "timestamp": "2020-01-18T15:55:00+00:00",
    "rawMessage": "KOKV 181555Z AUTO 16010G15KT 5SM UP OVC055 M02/M06 A3032 RMK AO2",
    "textDescription": "Unknown Precipitation",
    "icon": "https://api.weather.gov/icons/land/day/ovc?size=medium",
    "presentWeather": [
      {
        "intensity": null,
        "modifier": null,
        "weather": "unknown",
        "rawString": "UP"
      }
    ],
    "temperature": {
      "value": -2,
      "unitCode": "unit:degC",
      "qualityControl": "qc:V"
    },
    "dewpoint": {
      "value": -6,
      "unitCode": "unit:degC",
      "qualityControl": "qc:V"
    },
    "windDirection": {
      "value": 160,
      "unitCode": "unit:degree_(angle)",
      "qualityControl": "qc:V"
    },
    "windSpeed": {
      "value": 5.1,
      "unitCode": "unit:m_s-1",
      "qualityControl": "qc:V"
    },
    "windGust": {
      "value": 7.7,
      "unitCode": "unit:m_s-1",
      "qualityControl": "qc:S"
    },
    "barometricPressure": {
      "value": 102680,
      "unitCode": "unit:Pa",
      "qualityControl": "qc:V"
    },
    "seaLevelPressure": {
      "value": null,
      "unitCode": "unit:Pa",
      "qualityControl": "qc:Z"
    },
    "visibility": {
      "value": 8050,
      "unitCode": "unit:m",
      "qualityControl": "qc:C"
    },
    "maxTemperatureLast24Hours": {
      "value": null,
      "unitCode": "unit:degC",
      "qualityControl": null
    },
    "minTemperatureLast24Hours": {
      "value": null,
      "unitCode": "unit:degC",
      "qualityControl": null
    },
    "precipitationLastHour": {
      "value": null,
      "unitCode": "unit:m",
      "qualityControl": "qc:Z"
    },
    "precipitationLast3Hours": {
      "value": null,
      "unitCode": "unit:m",
      "qualityControl": "qc:Z"
    },
    "precipitationLast6Hours": {
      "value": null,
      "unitCode": "unit:m",
      "qualityControl": "qc:Z"
    },
    "relativeHumidity": {
      "value": 74.09027171838983,
      "unitCode": "unit:percent",
      "qualityControl": "qc:C"
    },
    "windChill": {
      "value": -7.478809390532206,
      "unitCode": "unit:degC",
      "qualityControl": "qc:V"
    },
    "heatIndex": {
      "value": null,
      "unitCode": "unit:degC",
      "qualityControl": "qc:V"
    },
    "cloudLayers": [
      {
        "base": {
          "value": 1680,
          "unitCode": "unit:m"
        },
        "amount": "OVC"
      }
    ]
  }
}
```

The image below shows "0.03" in the 1 Hour Precipitation (cm) field:

<img width="960" alt="Github Issue - 2020-01-20" src="https://user-images.githubusercontent.com/44888229/72747885-d647b680-3b83-11ea-81be-0d4a442b3970.png">

**Expected behavior**
Consistency between data returned from casual browsers of stations' observational history and API users.

**Environment**
These API calls were repeated in Firefox 71.0 (64-bit), the [API Web Service ](https://www.weather.gov/documentation/services-web-api#/)page "Try It Out" feature, `curl` command line utility (Ubuntu 18.04 WSL), and PowerShell `Invoke-WebRequest`.

**Comment by joshwalker9115** [comment]
This appears to be a timestamp misunderstanding on my part, the API responds with UTC, not the timezone of the station which you are requesting.

This request with -05:00 appended works:
`https://api.weather.gov/stations/KOKV/observations/2020-01-18T15:55:00-05:00`

It may be helpful to add a note on the [API Web Service ](https://www.weather.gov/documentation/services-web-api#/) page that all times returned are in UTC. Under specifications it notes that 

> "All times generated by the API are in ISO-8601 format."

I may be splitting hairs but maybe "All times generated by the API are in ISO-8601 format and default to UTC." is clearer.

Thanks!

**Comment by StephenClouse** [comment]
The times do not necessarily default to UTC. This is why we present all times in ISO8601 and include the time zone offset.

**Comment by joshwalker9115** [comment]
That makes sense, thanks for the clarification!
